enabled ed25519/KMU for b0 on nrf54l and Cut down configuration for B0

de-nordic opened the pull request:
WARNING: The PR has changed with addition of defaulting ed25519/KMU in b0. That commit was not there previously as the base PR used for this has been enabling such configurations.
The original purpose of the cut down was to fit the PSA hardware crypto, so it would not make sense to keep the PR unless we now enable the ed25519 here.


This configuration allows B0 to use PSA crypto while still being able to fit into FPROTECT area.

Only three top commits. 

~~Depends on https://github.com/nrfconnect/sdk-nrf/pull/20812~~

For CI to test from branch:
test_boot: testing_21310_b0_on_nrf54l

de-nordic committed:
bootloader: bl_crypto: Switch to PSA-Lite for B0

Configuration options that enable PSA lite with B0 NSIB.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

NordicBuilder commented:
The following west manifest projects have changed revision in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| zephyr | https://github.com/nrfconnect/sdk-zephyr/commit/9ec6585407fc97aedc799f500c2f6f9ae621b94b | https://github.com/nrfconnect/sdk-zephyr/pull/2637 | [nrfconnect/sdk-zephyr#2637/files](https://github.com/nrfconnect/sdk-zephyr/pull/2637/files) |

⛔ **DNM label due to: 1 project with PR revision**

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

NordicBuilder commented:
## CI Information

<sup>_To view the history of this post, clich the 'edited' button above_</sup>
Build number: 1
### Inputs:
#### Sources:
sdk-nrf: PR head: [9cfdd7da6ce064d7ea33e6579b600ceb2ad94b06](https://github.com/nrfconnect/sdk-nrf/pull/21310/commits/9cfdd7da6ce064d7ea33e6579b600ceb2ad94b06)
zephyr: PR head: [87f0a1453ccb04cd194a81a399fc3661245c5da1](https://github.com/nrfconnect/sdk-zephyr/pull/2637/commits/87f0a1453ccb04cd194a81a399fc3661245c5da1)
<details><summary> more details</summary>

sdk-nrf:
> PR head: [9cfdd7da6ce064d7ea33e6579b600ceb2ad94b06](https://github.com/nrfconnect/sdk-nrf/pull/21310/commits/9cfdd7da6ce064d7ea33e6579b600ceb2ad94b06)
> merge base: [6a99357fb00f3f468ad6396d58b324315718c315](https://github.com/nrfconnect/sdk-nrf/commit/6a99357fb00f3f468ad6396d58b324315718c315)
> target head (main): [50bf38201f565d885284b2bcff66a9a7d55d933d](https://github.com/nrfconnect/sdk-nrf/commit/50bf38201f565d885284b2bcff66a9a7d55d933d)
> [Diff](https://github.com/nrfconnect/sdk-nrf/compare/6a99357fb00f3f468ad6396d58b324315718c315..9cfdd7da6ce064d7ea33e6579b600ceb2ad94b06)

zephyr:
> PR head: [87f0a1453ccb04cd194a81a399fc3661245c5da1](https://github.com/nrfconnect/sdk-zephyr/pull/2637/commits/87f0a1453ccb04cd194a81a399fc3661245c5da1)
> merge base: [9ec6585407fc97aedc799f500c2f6f9ae621b94b](https://github.com/nrfconnect/sdk-zephyr/commit/9ec6585407fc97aedc799f500c2f6f9ae621b94b)
> target head (main): [e5d1ba3584e03caaa682406422c490617367badc](https://github.com/nrfconnect/sdk-zephyr/commit/e5d1ba3584e03caaa682406422c490617367badc)
> [Diff](https://github.com/nrfconnect/sdk-zephyr/compare/9ec6585407fc97aedc799f500c2f6f9ae621b94b..87f0a1453ccb04cd194a81a399fc3661245c5da1)
</details>

#### Github labels
| Enabled | Name | Description |
| -- | -- | -- |
|| ci-disabled | _Disable the ci execution_ | 
|| ci-all-test | _Run all of ci, no test spec filtering will be done_ | 
|| ci-force-downstream | _Force execution of downstream even if twister fails_ | 
|| ci-run-twister | _Force run twister_ | 
|| ci-run-zephyr-twister | _Force run zephyr twister_ | 

<details><summary> List of changed files detected by CI (30)</summary>

```
CODEOWNERS
subsys
│  ├── bootloader
│  │  ├── Kconfig
│  │  ├── bl_crypto
│  │  │  │ Kconfig
│  ├── nrf_security
│  │  ├── src
│  │  │  ├── core
│  │  │  │  ├── CMakeLists.txt
│  │  │  │  ├── Kconfig
│  │  │  │  ├── lite
│  │  │  │  │  ├── CMakeLists.txt
│  │  │  │  │  ├── psa_core_lite.c
│  │  │  │  │  │ psa_core_lite.h
│  │  │  ├── drivers
│  │  │  │  ├── cracen
│  │  │  │  │  ├── cracenpsa
│  │  │  │  │  │  ├── include
│  │  │  │  │  │  │  │ cracen_psa_key_ids.h
│  │  │  │  │  │  ├── src
│  │  │  │  │  │  │  ├── key_management.c
│  │  │  │  │  │  │  │ kmu.c
│  │  │  │ psa_crypto_driver_wrappers.c
sysbuild
│  │ Kconfig.secureboot
tests
│  ├── subsys
│  │  ├── nrf_security
│  │  │  ├── psa_core_lite
│  │  │  │  ├── CMakeLists.txt
│  │  │  │  ├── boards
│  │  │  │  │  ├── nrf54l15dk_nrf54l15_cpuapp.conf
│  │  │  │  │  │ nrf54l15pdk_nrf54l15_cpuapp.conf
│  │  │  │  ├── overlays
│  │  │  │  │  ├── ecdsa.conf
│  │  │  │  │  ├── eddsa.conf
│  │  │  │  │  ├── encrypt.conf
│  │  │  │  │  ├── lite.conf
│  │  │  │  │  │ rng.conf
│  │  │  │  ├── prj.conf
│  │  │  │  ├── src
│  │  │  │  │  │ main.c
│  │  │  │  │ testcase.yaml
west.yml
zephyr
│  ├── include
│  │  ├── zephyr
│  │  │  ├── sys
│  │  │  │  ├── util_internal.h
│  │  │  │  │ util_macro.h
│  ├── scripts
│  │  ├── checkpatch.pl
│  │  ├── ci
│  │  │  │ check_compliance.py
│  ├── tests
│  │  ├── unit
│  │  │  ├── util
│  │  │  │  │ main.c
```
</details>

### Outputs:

#### Toolchain
Version: 
Build docker image: 

#### Test Spec & Results: :white_check_mark: Success; :x: Failure; :orange_circle: Queued; :yellow_circle: Progress; :white_medium_square: Skipped; :warning: Quarantine
- :orange_circle: Toolchain
- :orange_circle: Build twister
- :orange_circle: Integration tests
   - :orange_circle: test-sdk-audio
   - :orange_circle: desktop52_verification
   - :orange_circle: test-fw-nrfconnect-boot
   - :orange_circle: test-fw-nrfconnect-apps
   - :orange_circle: test_ble_nrf_config
   - :orange_circle: test-fw-nrfconnect-ble_mesh
   - :orange_circle: test-fw-nrfconnect-ble_samples
   - :orange_circle: test-fw-nrfconnect-chip
   - :orange_circle: test-fw-nrfconnect-nfc
   - :orange_circle: test-fw-nrfconnect-nrf-iot_cloud
   - :orange_circle: test-fw-nrfconnect-nrf-iot_libmodem-nrf
   - :orange_circle: test-fw-nrfconnect-nrf-iot_serial_lte_modem
   - :orange_circle: test-fw-nrfconnect-nrf-iot_zephyr_lwm2m
   - :orange_circle: test-fw-nrfconnect-nrf-iot_samples
   - :orange_circle: test-fw-nrfconnect-nrf-iot_lwm2m
   - :orange_circle: doc-internal
   - :orange_circle: test-fw-nrfconnect-nrf-iot_thingy91
   - :orange_circle: test-fw-nrfconnect-nrf_crypto
   - :orange_circle: test-fw-nrfconnect-proprietary_esb
   - :orange_circle: test-fw-nrfconnect-rpc
   - :orange_circle: test-fw-nrfconnect-rs
   - :orange_circle: test-fw-nrfconnect-fem
   - :orange_circle: test-fw-nrfconnect-tfm
   - :orange_circle: test-fw-nrfconnect-thread
   - :orange_circle: test-sdk-find-my
   - :orange_circle: test-fw-nrfconnect-nrf-iot_mosh
   - :orange_circle: test-fw-nrfconnect-nrf-iot_positioning
   - :orange_circle: test-sdk-wifi
   - :orange_circle: test-low-level
   - :orange_circle: test-sdk-pmic-samples
   - :orange_circle: test-sdk-mcuboot
   - :orange_circle: test-sdk-dfu
   - :orange_circle: test-fw-nrfconnect-ps
   - :orange_circle: test-secdom-samples-public
   - :warning: test-fw-nrfconnect-fw-update



*Note: This message is automatically posted and updated by the CI*

de-nordic committed:
bootloader: Cut banner and events from configuration

Cuts NCS banner and events/notifications to further reduce size.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nvlsianpu reviewed (approved):
approved pluto's commits

nordicjm commented on subsys/bootloader/bl_crypto/Kconfig:
```diff
@@ -107,6 +108,15 @@ config SB_CRYPTO_PSA_ED25519
+	  This is copy of PSA_CORE choice from NRF Security Kconfig, as the
+	  only way to select an element from choice is to override default
+	  for it.
 endchoice
```
newline

de-nordic replied:
Done

nordicjm commented on subsys/bootloader/bl_crypto/Kconfig:
```diff
@@ -107,6 +108,15 @@ config SB_CRYPTO_PSA_ED25519
 	#	  Hardware implementation of RSASSA-PSS, 2048-bits.
 	#	  Not available yet.
 
+
```
```suggestion
```

Vge0rge commented on subsys/bootloader/Kconfig:
```diff
@@ -74,7 +74,8 @@ config PM_PARTITION_SIZE_B0_IMAGE
 	default FPROTECT_BLOCK_SIZE if SOC_SERIES_NRF91X || SOC_NRF5340_CPUAPP
 	default 0x3800 if SOC_NRF5340_CPUNET
-	default 0x7800 if SOC_NRF54L15_CPUAPP
+	default 0x9800 if SOC_NRF54L15_CPUAPP && !FPROTECT
```
Just for my own shake, why do we set a different size depending on the FPROTECT?

de-nordic replied:
If you disable FPROTECT you can fit more into b0, because fprotect can only protect 31k.
So, for example, if you want to enable a lot of stuff to debug, you may not fit in the FPROTECT area and need to disable it, then it automatically also brings the size of allowed code up.

Vge0rge reviewed (approved):
This PR basically includes PR 20812. I reviewed the last commits which set the configuration of the PSA_CORE_LITE and they seem OK to me.

de-nordic commented:
> This PR basically includes PR 20812. I reviewed the last commits which set the configuration of the PSA_CORE_LITE and they seem OK to me.

Yes. That PR is required for the change to work.

michalek-no committed:
lib: fprotect: Kconfig disable for B0 on nrf54L

temporarily diabled

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

de-nordic committed:
tests: bootloader: Temporarily disable nrf54l15 with b0

Testing requires upload of KMU keys, which is currently
not available.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
bootloader: Default to ed25519 and KMU for nrf54l b0

The commit defaults to usage of ed25519 signature with KMU,
when building b0 for nrf54l.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nvlsianpu commented on sysbuild/Kconfig.secureboot:
```diff
@@ -37,6 +37,7 @@ config SECURE_BOOT_APPCORE_SUPPORTED_HASH_NONE
 config SECURE_BOOT_APPCORE_SUPPORTED_SIGNATURE_HARDWARE
 	bool
 	default y if SECURE_BOOT_SIGNATURE_TYPE_ECDSA && (SOC_SERIES_NRF91X || SOC_NRF52840)
+	default y if SECURE_BOOT_SIGNATURE_TYPE_ED25519 && SOC_NRF54L15_CPUAPP
```
extend for enabling this on SOC_NRF54l05_CPUAPP and SOC_NRF54l10_CPUAPP
same in else places in that PR

nordicjm commented on samples/bootloader/prj.conf:
```diff
@@ -30,3 +30,10 @@ CONFIG_ASSERT=n
 # Avoid triggering IRQs from the RTC
 CONFIG_NRF_RTC_TIMER=n
+
+CONFIG_CLOCK_CONTROL_NRF=n
+CONFIG_NOTIFY=n
+CONFIG_ONOFF=n
+
+CONFIG_BOOT_BANNER=n
+CONFIG_NCS_BOOT_BANNER=n
```
should not be applying this for everything, add a `boards/<board>` Kconfig fragment, and it should have a file suffix because people expect the default one to all act the same way which is with logging

frkv reviewed (approved):
LGTM
